lib/web_ui/lib/src/ui/window.dart

```diff
@@ -638,6 +643,36 @@ abstract class Window {
   Brightness get platformBrightness => _platformBrightness;
   Brightness _platformBrightness = Brightness.light;
 
+  /// Reference to css media query that indicates the user theme preference on the web
+  final html.MediaQueryList _brightnessMediaQuery = html.window.matchMedia('(prefers-color-scheme: dark)');
+
+  /// A callback that is invoked whenever [_brightnessMediaQuery] changes value.
+  /// Updates the platformBrightness with the new user preferenceq
+  ///
+  ///  The param brightnessMediaQuery can be a MediaQueryListEvent or a MediaQueryList;
+  ///
+  void _brightnessMediaQueryChanged(brightnessMediaQuery) {
+      Brightness previousPlatformBrightness = _platformBrightness;
+      _platformBrightness = brightnessMediaQuery.matches ? Brightness.dark : Brightness.light;
+      if (previousPlatformBrightness != _platformBrightness &&
+          _onPlatformBrightnessChanged is VoidCallback)
```

Comment:
`_onPlatformBrightnessChanged != null` should be enough here. No need to check the exact type of the function.